libstuff/SHTTPSManager.cpp

```diff
@@ -239,3 +240,57 @@ bool SStandaloneHTTPSManager::_onRecv(Transaction* transaction)
     transaction->response = getHTTPResponseCode(transaction->fullResponse.methodLine);
     return false;
 }
+
+list<SStandaloneHTTPSManager::Transaction*> SStandaloneHTTPSManager::_httpsSendMultiple(const string& url, vector<SData>& sendRequests)
+{
+    // Synchronize dequeuing requests, and saving results.
+    int currentIndex = 0;
+
+    string host, path;
+    if (!SParseURI(url, host, path)) {
+        _activeTransactionList.push_front(_createErrorTransaction());
+        return _activeTransactionList;
+    }
+    if (!SContains(host, ":")) {
+        host += ":443";
+    }
+
+    // Our results go here.
+    list<SStandaloneHTTPSManager::Transaction*> currentSocketTransactions;
+
+    // This tries to create a socket on the correct port.
+    SX509* x509 = SX509Open(_pem, _srvCrt, _caCrt);
+    Socket* s = openSocket(host, x509);
+
+    size_t myIndex = 0;
+    SData myRequest;
+    while (true) {
```

Comment:
I know you copy pasted this from somewhere else, but this doesn't make sense here. There are no threads that might start on these out-of-order. This can just be:
```
for (auto& request : sendRequests)
```

And then you don't need `myIndex`, `myRequest` and `currentIndex`.

Reply:
Ah, good call.